lib/internal/modules/esm/loader.js

```diff
@@ -274,83 +369,12 @@ class DefaultModuleLoader {
 
   importMetaInitialize(meta, context) {
     importMetaInitializer ??= require('internal/modules/esm/initialize_import_meta').initializeImportMeta;
+    kResolveSync ??= require('internal/modules/esm/initialize_import_meta').kResolveSync;
     meta = importMetaInitializer(meta, context, this);
     return meta;
   }
 }
-ObjectSetPrototypeOf(DefaultModuleLoader.prototype, null);
-
-
-class CustomizedModuleLoader extends DefaultModuleLoader {
```

Comment:
I think this did not need to move: it's referenced inside `createModuleLoader`, so it will have already been parsed before a call to `createModuleLoader` runs.

If it's not actually required to move it, please put it back.

Reply:
I think there was a lint warning if I left it at the bottom… but I can try again!

Reply:
Indeed:

```sh
make lint-js
Running JS linter...

node/lib/internal/modules/esm/loader.js
  260:26  error  'CustomizedModuleLoader' was used before it was defined  no-use-before-define
```

I'll leave it as it stands for now while we review for legibility reasons though.

Reply:
Hmm 😑

Fair, but I think there'll be a way around that, and it would make for a much nicer diff if it was not moved. I'm unresolving to keep the reminder for myself.

Reply:
I could add `let DefaultCustomizedModuleLoader;` or something at the top of the file and then

```js
DefaultCustomizedModuleLoader = CustomizedModuleLoader
```

at this line. I could also `// eslint-disable-next-line` or some such.